internal/resource/url.go

```diff
@@ -283,6 +297,72 @@ func (f *Fetcher) FetchFromOEM(u url.URL, dest FetchWriter) ([]byte, error) {
 	return f.copyAndHash(dest, src)
 }
 
+// FetchFromS3 gets data from an S3 bucket as described by u and writes it into
+// dest, returning a hash (if one is set) or an error. It will attempt to
+// acquire IAM credentials from the EC2 metadata service, and if this fails will
+// attempt to fetch the object with anonymous credentials.
+func (f *Fetcher) FetchFromS3(u url.URL, dest FetchWriter) ([]byte, error) {
+	if f.AWSRegion == "" {
+		return nil, ErrAWSRegionNotSet
+	}
+	sess, err := session.NewSession(&aws.Config{
```

Comment:
This session is created regardless of the region even though it's only used on EC2.